src/unix/tcp.c

```diff
@@ -225,9 +227,33 @@ static int uv__is_ipv6_link_local(const struct sockaddr* addr) {
 
 static int uv__ipv6_link_local_scope_id(void) {
   struct sockaddr_in6* a6;
+  int rv;
+  // AIX path fallback to use uv_interface_addresses
+#if defined (_AIX) && !defined(__PASE__)
+  uv_interface_address_t* interfaces;
+  int count, i;
+
+  if (uv_interface_addresses(&interfaces, &count)) {
+    return 0;
+  }
+
+  rv = 0;
+
+  for (i = 0; i < count; i++) {
+    if (uv__is_ipv6_link_local((const struct sockaddr*) &interfaces[i].address.address6)) {
+      if (&interfaces[i].address.address6 != NULL) {
+        a6 = &interfaces[i].address.address6;
+        rv = a6->sin6_scope_id;
+      }
+      break;
+    }
+  }
```

Comment:
```suggestion
  for (ifa = interfaces; ifa != &interfaces[count]; ifa++) {
    if (uv__is_ipv6_link_local((struct sockaddr*) &ifa->address)) {
      rv = ifa->address.address6.sin6_scope_id;
      break;
    }
  }
```

Reply:
The errors suggest you have `uv_interface_address_t` (value type), not `uv_interface_address_t*` (pointer type)